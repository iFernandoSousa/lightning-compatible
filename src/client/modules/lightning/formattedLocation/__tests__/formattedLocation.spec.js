/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/formattedLocation';

const createComponent = attributes => {
    const element = createElement('c-formatted-location', {
        is: Element
    });

    Object.assign(element, attributes);
    document.body.appendChild(element);
    return element;
};

describe('c-formatted-location', () => {
    it('default', () => {
        const element = createComponent({
            longitude: 100.0,
            latitude: 90
        });

        expect(element).toMatchSnapshot();
    });

    it('normal longitude and latitude', () => {
        const element = createComponent({
            longitude: 122.222,
            latitude: 22.222
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('too big longitude', () => {
        const element = createComponent({
            longitude: 182.222,
            latitude: 22.222
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('too big latitude', () => {
        const element = createComponent({
            longitude: 182.222,
            latitude: 99.222
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('invalid longitude and latitude', () => {
        const element = createComponent({
            longitude: '122.222sss',
            latitude: 22.222
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('missing longitude and latitude', () => {
        const element = createComponent({
            longitude: null,
            latitude: 22.222
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
