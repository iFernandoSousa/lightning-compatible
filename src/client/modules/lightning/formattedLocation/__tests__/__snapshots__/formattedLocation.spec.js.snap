// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-formatted-location default 1`] = `
<c-formatted-location>
  #shadow-root(open)
    90
    , 
    100
</c-formatted-location>
`;

exports[`c-formatted-location invalid longitude and latitude 1`] = `
<c-formatted-location>
  #shadow-root(open)
</c-formatted-location>
`;

exports[`c-formatted-location missing longitude and latitude 1`] = `
<c-formatted-location>
  #shadow-root(open)
</c-formatted-location>
`;

exports[`c-formatted-location normal longitude and latitude 1`] = `
<c-formatted-location>
  #shadow-root(open)
    22.222
    , 
    122.222
</c-formatted-location>
`;

exports[`c-formatted-location too big latitude 1`] = `
<c-formatted-location>
  #shadow-root(open)
</c-formatted-location>
`;

exports[`c-formatted-location too big longitude 1`] = `
<c-formatted-location>
  #shadow-root(open)
</c-formatted-location>
`;
